src/ExpressionEvaluator/CSharp/Test/ResultProvider/ValueFormattingTests.cs

```diff
@@ -214,8 +214,8 @@ public void String()
             Assert.Equal("\"\\uffff\"", FormatValue(s, useHexadecimal: true));
 
             string multiByte = "\ud83c\udfc8";
-            Assert.Equal(string.Format(format, "🏈"), FormatValue(multiByte));
-            Assert.Equal(string.Format(format, "🏈"), FormatValue(multiByte, useHexadecimal: true));
+            Assert.Equal(string.Format(format, "\\ud83c\\udfc8"), FormatValue(multiByte));
+            Assert.Equal(string.Format(format, "\\ud83c\\udfc8"), FormatValue(multiByte, useHexadecimal: true));
```

Comment:
@tannergooding There would be two problems leaving surrogate pairs in source:
- The direct translation of Unicode surrogates to UTF-8 fails, which is an obstacle to writing source into a UTF-8 encoded file. See http://unicodebook.readthedocs.io/issues.html#strict-utf8-decoder
- Some categories of Unicode code points are not valid in a string literal (e.g. control characters, formatting characters, etc) and must be escaped. But we do not have platform APIs that help us determine the Unicode category of Unicode code points that are represented by surrogate pairs. So we have no way of knowing whether escaping is required. But escaping is always correct.